staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -229,13 +231,13 @@ func (d *validatingDispatcher) callHook(ctx context.Context, h *v1.ValidatingWeb
 	}
 
 	if err := r.Do(ctx).Into(response); err != nil {
-		return &webhookutil.ErrCallingWebhook{WebhookName: h.Name, Reason: err}
+		return &webhookutil.ErrCallingWebhook{WebhookName: h.Name, Reason: fmt.Errorf("failed to call webhook: %w", err), Status: apierrors.NewBadRequest("Error executing webhook request")}
 	}
 	trace.Step("Request completed")
 
 	result, err := webhookrequest.VerifyAdmissionResponse(uid, false, response)
 	if err != nil {
-		return &webhookutil.ErrCallingWebhook{WebhookName: h.Name, Reason: err}
+		return &webhookutil.ErrCallingWebhook{WebhookName: h.Name, Reason: fmt.Errorf("received invalid webhook response: %w", err), Status: apierrors.NewBadRequest("error verifying webhook response")}
```

Comment:
Not all of these are badRequest. I think this one should be NewClientErrorReporter. Read through the options here https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors?utm_source=godoc and pick the most appropriate one for each case

Reply:
Changed this one to a NewServiceUnavailable error. NewClientErrorReporter response is not ideal since that is of type ErrorReporter and we need it to be of type type StatusError.